localstack/services/cloudformation/models/ec2.py

```diff
@@ -147,9 +147,16 @@ def fetch_state(self, stack_name, resources):
         gw_id = self.resolve_refs_recursively(stack_name, props.get("GatewayId"), resources)
         route_tables = client.describe_route_tables()["RouteTables"]
         route_table = ([t for t in route_tables if t["RouteTableId"] == table_id] or [None])[0]
+        subnet_id = props.get("SubnetId")
```

Comment:
Since `SubnetId` may potentially be a reference to a property from another resource or stack parameter (something like `!Ref: SubnetA` below), we should use `resolve_refs_recursively` here to resolve it. Can we change this to:

```
subnet_id = self.resolve_refs_recursively(stack_name, props.get("SubnetId"), resources)
```

This doesn't seem to surface in your tests below, but it is still possible that the value needs to be resolved here, based on how the underlying CloudFormation engine is resolving the props. (it may surface on a stack update operation, actually)

Reply:
Thank you for quick review, I've updated the code with this recommendation.